internal/job/executor.go

```diff
@@ -1717,48 +1718,59 @@ func (e *Executor) runPostCommandHooks(ctx context.Context) error {
 }
 
 // CommandPhase determines how to run the build, and then runs it
-func (e *Executor) CommandPhase(ctx context.Context) (error, error) {
+func (e *Executor) CommandPhase(ctx context.Context) (hookErr error, commandError error) {
+	var preCommandErr error
+
 	span, ctx := tracetools.StartSpanFromContext(ctx, "command", e.ExecutorConfig.TracingBackend)
-	var err error
-	defer func() { span.FinishWithError(err) }()
+	defer func() {
+		span.FinishWithError(hookErr)
```

Comment:
The span includes the command error if it is not nil when this method is returned. But, if there is both a post-command hook error and a command error, the post-command hook error would have finished the span. This seems like a bug. But as I'm aiming for a no-op refactor. I don't want to address that here.